src/store/dappChainStore.ts

```diff
@@ -273,28 +262,29 @@ export default {
         throw new Error('No Private Key, Login again')
       }
       const network = state.networkId
+      const domainType = getDomainType()
       let user
-
-      let dposConstructor
-
-      if (['dev', 'local'].includes(getDomainType())) {
-        dposConstructor = DPOSUser.createEthSignMetamaskUserAsync
-      } else {
-        dposConstructor = DPOSUser.createMetamaskUserAsync
-      }
-
       try {
-        user = await dposConstructor(
-        payload.web3,
-        state.currentChain.dappchainEndpoint,
-        privateKeyString,
-        network,
-        GW_ADDRESS || GatewayJSON.networks[network].address,
-        LOOM_ADDRESS || LoomTokenJSON.networks[network].address
-        );
+        if (domainType === 'dev' || domainType === 'local') {
+          user = await DPOSUserV3.createEthSignMetamaskUserAsync({
+            web3: payload.web3,
+            dappchainEndpoint: state.chainUrls[state.networkId],
+            chainId: network,
+            gatewayAddress: GW_ADDRESS || GatewayJSON.networks[network].address,
+            version: 2
+          });
+        } else {
+          user = await DPOSUserV3.createMetamaskUserAsync({
+            web3: payload.web3,
+            dappchainEndpoint: state.chainUrls[state.networkId],
+            dappchainPrivateKey: privateKeyString,
+            chainId: network,
+            gatewayAddress: GW_ADDRESS || state.currentChain["gatewayAddress"],
+            version: 2
```

Comment:
```suggestion
            version: 1
```

Should be 1 for single-sig gateways. Eventually we'll transition to using 2 once the multisigs are deployed on mainnet.